src/plugins/credential_management/server/routes/create.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { IRouter } from '../../../../core/server';
+import { schema } from '@osd/config-schema';
+import { createHandler } from '../credential';
+
+export function registerCreateRoute(router: IRouter) {
+    router.post(
+        {
+          path: '/api/credential_management/create',
+          validate: {
+            body: schema.object({
+                credential_name: schema.string(),
+                credential_type: schema.oneOf([
+                  schema.literal('basic_auth'),
```

Comment:
i prefer to use `username_password_credential` here instead of basic_auth


basic auth is authentication type, not credential type. 

In basic HTTP authentication, a request contains a header field in the form of Authorization: Basic <credentials>, where credentials is the Base64 encoding of ID and password joined by a single colon :



![image](https://user-images.githubusercontent.com/550437/176568620-f1a23bbf-7dae-4e9d-9856-3be93cf2f4fe.png)